common-ts/src/types/UIEnv.ts

```diff
@@ -0,0 +1,58 @@
+import { DriftEnv } from '@drift-labs/sdk';
+import { Opaque } from '.';
+
+// Define UIEnvKey as an opaque type
+export type UIEnvKey = Opaque<'UIEnvKey', string>;
+
+/**
+ * Utility class to handle environment configs in the Drift UI. Note that these environments don't exactly align with the SDK environment (SDK can only be mainnet or devnet, but UI can be devnet, staging, mainnet) so there is a `sdkEnv` getter utility to convert the UI env to SDK env (UI staging => SDK mainnet).
+ */
+export class UIEnv {
+	constructor(readonly env: DriftEnv | 'staging') {}
+
+	static createMainnet() {
+		return new UIEnv('mainnet' as DriftEnv);
+	}
+
+	static createDevnet() {
+		return new UIEnv('devnet' as DriftEnv);
+	}
+
+	static createStaging() {
+		return new UIEnv('staging');
```

Comment:
Isn't staging kind of a separate thing from devnet vs mainnet? Tbh I'm not totally sure how we prefer to represent it. I was thinking this is like a solana network vs drift deployment env thing.

Reply:
I guess this is in the context of the data API? I see now.

Would be cool if we standardized all our frontends+backends on what environments we cover. I feel like there is a lot of discrepancy. Like swift server has `master` and `mainnet`. Just thinking out lout.  This probably works for data API though!

Reply:
yeah I actually agree that standardising how we communicate/configure environments across the different stacks is a good thing to do!! 

similarly what you raised above, what should be the proper delineation between "mainnet"/"mainnet-beta"/"production"? I don't think we've tried to tackle this at all but it is confusing